captcha-plugin-draft.txt

```diff
@@ -313,6 +324,9 @@ Appendix D: Security Considerations [SECURITY]
 
     [TODO: We've thought about this, but we're far from experts in anonymity.
     Review is badly needed!]
+    [Alex: Is it possible to verify that each plugin doesn't store a unique key
+    for each partcular browser? Otherwise the same attack could be carried out
+    regardless if CLoudflare keeps all the public keys.]
```

Comment:
The idea is to rely on the existing TBB [build infrastructure](https://blog.torproject.org/blog/deterministic-builds-part-two-technical-details). It should be quite hard to deliver a unique Tor Browser to each user without detection.